runtime/shared_common/shrinit.cpp

```diff
@@ -4303,6 +4303,14 @@ j9shr_shutdown(J9JavaVM *vm)
 		if (config->jclCacheMutex) {
 			omrthread_monitor_destroy(config->jclCacheMutex);
 		}
+
+		if (compareAndSwapUDATA((uintptr_t *)&config->romToRamHashTable, (uintptr_t)romToRamHashTable, (uintptr_t)NULL)) {
+			config->romToRamHashTableMutex = NULL;
+			omrthread_rwmutex_enter_write(romToRamHashTableMutex);
```

Comment:
The local variables `romToRamHashTable` and `romToRamHashTableMutex` should be declared and initialized here; the return value should be compared explicitly with `NULL` and the long line folded:
```c
		J9HashTable *romToRamHashTable = config->romToRamHashTable;
		if (NULL != compareAndSwapUDATA(
				(uintptr_t *)&config->romToRamHashTable,
				(uintptr_t)romToRamHashTable,
				(uintptr_t)NULL)
		) {
			omrthread_rwmutex_t romToRamHashTableMutex = config->romToRamHashTableMutex;
			config->romToRamHashTableMutex = NULL;
			omrthread_rwmutex_enter_write(romToRamHashTableMutex);
```

Reply:
Shouldn't `J9HashTable *romToRamHashTable` be declared at the beginning of the block where it is now?